src/test/java/uk/gov/nationalarchives/pdi/step/jena/serializer/JenaSerializerStepIT.java

```diff
@@ -0,0 +1,192 @@
+
+        // Would throw if var was not resolved
+        assertTrue(rowProcessed);
+    }
+
+    @Test
+    public void resolves_filename_variable() throws KettleException {
+        // TODO(SL): Use temp folder instead of random file
+        final String expectedFilename = String.valueOf(new Random().nextInt());
+
+        final JenaSerializerStepMeta meta = getMeta();
+        final StepDataInterface data = getData();
+        final StepMockHelper helper = mockHelper();
+        final JenaSerializerStep step = mockStep(helper);
+
+        meta.setJenaModelField("model");
+
+        step.setVariable("filenameVar", expectedFilename);
+        meta.setFileDetail(
+                new JenaSerializerStepMeta.FileDetail() {{
+                    filename = "${filenameVar}";
+                }});
+
+        // First pass create model, second serialise
+        step.processRow(meta, data);
+        step.processRow(meta, data);
+
+        File f = new File(expectedFilename);
+
+        assertTrue(f.exists() && !f.isDirectory());
+    }
+
+    @Test
+    public void resolves_serialization_format_variable() throws KettleException {
+        final JenaSerializerStepMeta meta = getMeta();
+        final StepDataInterface data = getData();
+        final StepMockHelper helper = mockHelper();
+        final JenaSerializerStep step = mockStep(helper);
+
+        meta.setJenaModelField("model");
+        meta.setFileDetail(new JenaSerializerStepMeta.FileDetail());
+
+        step.setVariable("serialisationFormatVar", "RDF/XML");
+        meta.setSerializationFormat("${serialisationFormatVar}");
+
+        // First pass create model, second serialise
+        step.processRow(meta, data);
+        final boolean rowProcessed = step.processRow(meta, data);
+
+        // Would throw if var was not resolved
+        assertFalse(rowProcessed);
+    }
+
+    private StepDataInterface getData() {
+        return new JenaSerializerStepData() {{
+            init();
+        }};
+    }
+
+    private static JenaSerializerStepMeta getMeta() {
+        return new JenaSerializerStepMeta() {{
+            setDefault();
+        }};
+    }
+
+    private static StepMockHelper mockHelper() {
+        final StepMockHelper helper = new StepMockHelper<>("Serialize Jena Model", JenaSerializerStepMeta.class, JenaSerializerStepData.class);
+
+        when(helper.logChannelInterfaceFactory.create(any(), any(LoggingObjectInterface.class))).thenReturn(helper.logChannelInterface);
+        when(helper.trans.isRunning()).thenReturn(true);
+
+        return helper;
+    }
+
+    private JenaSerializerStep mockStep(StepMockHelper helper) throws KettleException {
+        final JenaSerializerStep step = Mockito.spy(new JenaSerializerStep(helper.stepMeta, helper.stepDataInterface, 0, helper.transMeta, helper.trans));
+
+        step.setRowHandler(
+                new StaticDataRowHandler(
+                        new ArrayList() {{

```

Comment:
1. Please do not initialize objects in this way! `{{` should be avoided.

2. You are missing the generic typing specification.

Instead please use the longer:
```
final List<Object[]> list = new ArrayList<>();
list.add(...)
```

or instead:
```
final List<Object[]> list = Arrays.asList(...)
```